### PR TITLE
Remove redundant module version related code. Add aodncore version to…

### PR DESCRIPTION
**Pull request by lwgordonimos**
… module versions dict.

### CODE REVIEW

File: aodncore/pipeline/handlerbase.py [hunk 1]
```diff
@@ -429,6 +421,19 @@ def instance_working_directory(self):
         """
         return self._instance_working_directory
 
+    @property
+    def module_versions(self):
+        """Read-only property to access module versions
+
+        :return: module version strings for aodncore, aodndata and compliance checker modules
```
Comment (mhidas) [inline aodncore/pipeline/handlerbase.py]: I don't see any mention of the aodndata version here. Is it possible to include it?

**Comment by mhidas** [comment]
@lwgordonimos I had a question about including the aodndata version (see above). If that's easy, let's include it, otherwise shall we just merge this?

**Comment by lwgordonimos** [comment]
@mhidas the short answer is that it's not easy to do *correctly*.

I think the one way dependency aodndata <- aodncore is an important boundary to maintain, so any direct imports are a bad idea. 

The other option is to advertise things as an entry point (in the same way as the handlers/dest path functions).

The only other option I could brainstorm would be to add an intermediate base class in the aodndata package which all the handlers inherit from, but I'm less keen on that, especially just for printing a version number!

So I think it might be best to hold off on merging this, until I've had a look at the entry points mechanism. No point merging with the intention to immediately change it anyway!

**Comment by lwgordonimos** [comment]
@mhidas I've updated the PR so that it uses entry points. This will allow us to update the setup.py in aodndata to advertise it's version, and it will be included in the module versions dict automagically.

**Comment by mhidas** [comment]
Nice one, thanks!

**Pull request merged**